clang/lib/CIR/CodeGen/CIRGenModule.cpp

```diff
@@ -67,7 +67,8 @@ void CIRGenModule::emitGlobal(clang::GlobalDecl gd) {
       return;
     }
   } else {
-    errorNYI(global->getSourceRange(), "global variable declaration");
+    const auto *vd = cast<VarDecl>(global);
+    assert(vd->isFileVarDecl() && "Cannot emit local var decl as global");
```

Comment:
This is going to result in an 'unused variable' warning in release mode.  So inline hte 'cast' into the assert, or put a `(void)vd` at one point.